src/Parsers/ASTSelectQuery.cpp

```diff
@@ -474,4 +478,54 @@ void ASTSelectQuery::setFinal() // NOLINT method can be made const
     tables_element.table_expression->as<ASTTableExpression &>().final = true;
 }
 
+bool ASTSelectQuery::hasQueryParameters() const
+{
+    std::queue<ASTPtr> queue;
+    queue.push(this->clone());
+
+    while (!queue.empty())
+    {
+        auto ast = queue.front();
+        queue.pop();
+
+        if (ast->as<ASTQueryParameter>())
+            return true;
+
+        for (const auto & child : ast->children)
+            queue.push(child);
+    }
+    return false;
+}
+
+NameToNameMap ASTSelectQuery::getQueryParameterValues() const
+{
+    NameToNameMap parameter_values;
+    std::queue<ASTPtr> queue;
+    queue.push(this->clone());
```

Comment:
addressed it by using this->where() instead of clone().
About the test, do you mean this , 
`CREATE VIEW v3 AS SELECT * FROM Catalog WHERE Price={price:UInt64} AND Quantity=3;`
If yes then its added, if not, could you please clarify